training/deepspeech_training/train.py

```diff
@@ -321,8 +321,24 @@ def get_tower_results(iterator, optimizer, dropout_rates):
                     # Retain tower's avg losses
                     tower_avg_losses.append(avg_loss)
 
+                    train_vars = tf.get_collection(tf.GraphKeys.TRAINABLE_VARIABLES)
+
+                    # Filter out layers if we want to freeze some
+                    if FLAGS.freeze_source_layers > 0:
+                        filter_vars = drop_freeze_number_to_layers(FLAGS.freeze_source_layers, "freeze")
+                        new_train_vars = list(train_vars)
```

Comment:
Is there a reason not to build up `new_train_vars` from empty, something like:
```python
new_train_vars = []
for tv in train_vars:
    if tv.name not in filter_vars:
        new_train_vars.append(tv)
train_vars = new_train_vars
```

Reply:
Seemed more intuitive, we want to train all except the filtered layers.
Your example doesn't work by the way, because the `filter_vars` contain names like `layer_1` and `train_vars` have the full layer name  `layer_1:dense:0`

Reply:
Hence the "something like" :) -- You could of course use `find()` or something like that. I have no particularly strong feeling about it, but err on the side of simplicity.